crates/core/src/client/tests/mod.rs

```diff
@@ -0,0 +1,258 @@
+
+    let doc = client.document().expect("Failed to get document");
+    let expected = r#"
+<Group id="flash-group" />
+<VStack>
+    <Text>
+        second_page
+    </Text>
+    <NavigationLink id="Next" destination="/nav/next">
+        <Text>
+            NEXT
+        </Text>
+    </NavigationLink>
+</VStack>"#;
+    assert_doc_eq!(expected, doc.to_string());
+}
+
+#[tokio::test]
+async fn test_back_and_forward_navigation() {
+    let url = format!("http://{HOST}/nav/first_page");
+    let mut config = LiveViewClientConfiguration::default();
+    config.format = Platform::Swiftui;
+
+    let client = LiveViewClient::initial_connect(config, url)
+        .await
+        .expect("Failed to create client");
+
+    // Verify initial page
+    let doc = client.document().expect("Failed to get document");
+    let expected = r#"
+<Group id="flash-group" />
+<VStack>
+    <Text>
+        first_page
+    </Text>
+    <NavigationLink id="Next" destination="/nav/next">
+        <Text>
+            NEXT
+        </Text>
+    </NavigationLink>
+</VStack>"#;
+    assert_doc_eq!(expected, doc.to_string());
+
+    let second_url = format!("http://{HOST}/nav/second_page");
+    client
+        .navigate(second_url, Default::default())
+        .await
+        .expect("Failed to navigate");
+
+    let doc = client.document().expect("Failed to get document");
+    let expected = r#"
+<Group id="flash-group" />
+<VStack>
+    <Text>
+        second_page
+    </Text>
+    <NavigationLink id="Next" destination="/nav/next">
+        <Text>
+            NEXT
+        </Text>
+    </NavigationLink>
+</VStack>"#;
+    assert_doc_eq!(expected, doc.to_string());
+
+    let third_url = format!("http://{HOST}/nav/third_page");
+    client
+        .navigate(third_url, Default::default())
+        .await
+        .expect("Failed to navigate");
+
+    assert!(client.can_go_back(), "Back navigation impossible");
+
+    client.back(None).await.expect("Failed to navigate back");
+
+    // Verify we're back on second page
+    let doc = client.document().expect("Failed to get document");
+    let expected = r#"
+<Group id="flash-group" />
+<VStack>
+    <Text>
+        second_page
+    </Text>
+    <NavigationLink id="Next" destination="/nav/next">
+        <Text>
+            NEXT
+        </Text>
+    </NavigationLink>
+</VStack>"#;
+    assert_doc_eq!(expected, doc.to_string());
+
+    assert!(client.can_go_forward(), "Forward navigation impossible");
+    client
+        .forward(None)
+        .await
+        .expect("Failed to navigate forward");
+
+    let doc = client.document().expect("Failed to get document");
+    let expected = r#"
+<Group id="flash-group" />
+<VStack>
+    <Text>
+        third_page
+    </Text>
+    <NavigationLink id="Next" destination="/nav/next">
+        <Text>
+            NEXT
+        </Text>
+    </NavigationLink>
+</VStack>"#;
+    assert_doc_eq!(expected, doc.to_string());
+}
+
+#[tokio::test]
+async fn thermostat_click() {
```

Comment:
This playback test is a really good summary of the sum total changes. If you are reviewing this spells out how the new changes are expected to be used.